### PR TITLE
Support `IF NOT EXISTS` when create index

### PR DESCRIPTION
**Pull request by liberwang1013**
## PR Info
This PR add support `IF NOT EXISTS` when create index

## Adds

- [x] support `IF NOT EXISTS` when create index in postgres & sqlite

### CODE REVIEW

File: src/index/create.rs [hunk 1]
```diff
@@ -129,9 +130,16 @@ impl IndexCreateStatement {
             primary: false,
             unique: false,
             index_type: None,
+            if_not_exists: false,
         }
     }
 
+    /// Create index if index not exists
```
Comment (ikrivosheev) [inline src/index/create.rs:137]: I think is better add doc tests instead `tests`. It will be like examples

**Review by ikrivosheev — COMMENTED** [review]
@liberwang1013 thank you for PR! Some comments.

File: src/index/create.rs (hunk 1 above)
Comment (liberwang1013) [reply src/index/create.rs:161]: already add doc tests

**Review by ikrivosheev — APPROVED** [review]
@liberwang1013 thank you. LGTM!

**Review by tyt2y3 — APPROVED** [review]
Awesome

**Comment by tyt2y3** [comment]
Thank you all

**Pull request merged**